vulture/core.py

```diff
@@ -461,14 +501,25 @@ def _define(
         confidence=DEFAULT_CONFIDENCE,
         ignore=None,
     ):
+        def ignored(lineno):
+            if (ignore and ignore(self.filename, name)) or _match(
+                name, self.ignore_names
+            ):
+                return True
+
+            # Check if the reported line is annotated with "# noqa".
+            return (
+                lineno in self.noqa_matches[ERROR_CODES[typ]]
+                or lineno in self.noqa_matches["all"]
+            )
```

Comment:
This can become a function `noqa.ignore_line(noqa_matches, lineno, typ)` and then we can add it to the if-statement above.